GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.utility;
+
+public final class Preconditions {
+    
+    /**
+     * The goal of this class is to improve the readability and to provide some common checking method to improve 
+     * the robustness of GVRF. But in some cases, this may be a significant performance cost. Try to use and further
+     * extend this class carefully with concern on performance issue.
+     * 
+     * Actually, one purpose of this class may help us avoid from using GVRJniException which is not a proper way to 
+     * throw/simulate an exception with JNI crash.
+     */
+    
+
+    public static final String INPUT_TYPE_ARRAY = "Input array";
+    public static final String INPUT_TYPE_VECTOR = "Input vector";
+    public static final String INPUT_TYPE_NORMALS = "Input normals";
+    public static final String INPUT_TYPE_TRIANGLE = "Input triangle";
+    public static final String INPUT_TYPE_VERTICES = "Input vertices";
+    public static final String INPUT_TYPE_TEXCOORDS = "Input texture coordinates";
+
+    /**
+     * A simple check to see if the object is non-null or otherwise the function will throw an
+     *  IllegalArgumentException.
+     *  
+     * @param object
+     *          the parameter object can be any instance of any class even an array.
+     * @param dataMessage
+     *          To point out the type of data, for the use when we try to throw an exception.
+     *          Type may not refer to the class name but a wider concept of class type such as vertices, normals.
+     * @throws IllegalArgumentException if the object is null pointer.
+     */
+    public static void checkNotNull(Object object, String dataMessage) {
+        if (object == null) {
+            throw Exceptions.IllegalArgument("%s can't be a null pointer.",
+                    dataMessage);
+        }
+    }
+    
+    /**
+     * We can see that there is a wide use of key-value pair for the parameters which are used in shader.
+     * This function is to ensure that the key is meaningful.
+     * @param key
+     *          Key in the key-value pair.
+     *          
+     * @throws IllegalArgumentException if the key is null or empty.
+     */
+    public static void checkKeyNotNullOrEmpty(String key){
+        if(key == null || key.trim().isEmpty()){
```

Comment:
Even if you create a private method, you still need to scan the string for whitespaces, something similar to trim().
Considering code readability, using String.isEmpty() can be still a good way.